worker.go

```diff
@@ -11,14 +11,15 @@ import (
 
 // Worker is the doodad that does work.
 type Worker struct {
-	id       int
-	WorkChan chan UDPMetric
-	QuitChan chan struct{}
-	metrics  int64
-	mutex    *sync.Mutex
-	stats    *statsd.Client
-	logger   *logrus.Logger
-	wm       WorkerMetrics
+	id        int
+	WorkChan  chan UDPMetric
+	QuitChan  chan struct{}
+	processed int64
```

Comment:
Does it make sense to keep this and `imported` perhaps in the `WorkerMetrics` so that they get implicitly reset and you don't have to bother doing that? It also keeps all the state in `WorkerMetrics`… (I'm only partway through this review, tho, so i may be prematurely giving an opinion.)